src/core/utils.ts

```diff
@@ -176,6 +176,46 @@ export async function queryInBatches<T>(
   };
 }
 
+/**
+ * Executes a query function repeatedly until continuation token is null i.e. all data is retrieved, adhering to the specified
+ * batch query configuration for maximum requests per second and items per request.
+ *
+ * @template T - The type of the data being queried.
+ * @param queryRecord - A function that performs the query. It takes the maximum number of items
+ *                      to retrieve (`take`) and an optional continuation token, and returns a
+ *                      promise that resolves to a `QueryResponse<T>`.
+ * @param config - The batch query configuration, including:
+ *   - `maxTakePerRequest`: The maximum number of items to retrieve per request.
+ *   - `requestsPerSecond`: The maximum number of requests to make per second.
+ * @returns A promise that containing all retrieved data
+ */
+export async function queryUntilComplete<T>(
```

Comment:
The primary concerns I have are a) whether or not there should be explicit error handling in this function, and b) this function allows for an unbounded number of items to be retrieved. While it looks like this utility method was put in place initially to retrieve users, which shouldn't have a huge number, this mechanism is still meant to be general-purpose (rightly so), but as a result can certainly run into issues when used for certain things.